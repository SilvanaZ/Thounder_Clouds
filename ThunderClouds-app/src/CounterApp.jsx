import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    // console.log(event);
    // setCounter (counter + 2);
    setCounter((c) => c + 2);
  };

  const handleDecrease = () => setCounter(counter - 2);

  const handleReset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleDecrease}>-1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

/* <button onClick={(event) => handleAdd(event, "Hola Ro")}>
        clickeame
      </button>   -> esto va en el RETURN

      const handleAdd = (event, newValue) => {
  console.log(newValue); ->SE PIDE QUE VAYA FUERA DE LA FUNCION PERO LA PRUEBA FUE DENTRO DE FUNCION Y ANTES DEL RETURN
};*/
