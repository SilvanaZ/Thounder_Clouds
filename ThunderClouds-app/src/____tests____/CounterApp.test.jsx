import { CounterApp } from "../CounterApp";
import { fireEvent, render, screen } from '@testing-library/react'

describe('Pruebas en CounterApp', () => {
    const initialValue = 10;
    test("tiene que coincidir con el snapshot", () => {

        const { container } = render(<CounterApp value={initialValue} />);
        expect(container).toMatchSnapshot();

    });
    test('debe mostrar valor inicial de 100 <CounterApp value={100} />', () => {
        render(<CounterApp value={100} />);
        expect(screen.getByText(100)).toBeTruthy();

        //expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain('100')
    });
    test('debe de incrementar con el boton +1', () => {

        render(<CounterApp value={initialValue} />);
        //fireEvent.click(screen.getByText('+1'))
        expect(screen.getByText('11')).toBeTruthy();
    })
});
