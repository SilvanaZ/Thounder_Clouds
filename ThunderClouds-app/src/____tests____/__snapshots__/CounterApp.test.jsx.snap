// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp tiene que coincidir con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
